### PR TITLE
refactor: architecture and media type selection

### PR DESCRIPTION
**Pull request by LeoDevSecOps**
- Refactored architecture as Fastapi recommand
- Added a little feature to select media_type in query

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -20,21 +22,22 @@ async def read_root():
...
-async def download_audio(url: Optional[str] = None, playlist: Optional[bool] = False):
+async def download_audio(url: Optional[str] = None, playlist: Optional[bool] = False, media_type: Optional[str] = 'webm'):
     current_download_dir = os.path.join(DOWNLOAD_DIR, slugify(url))
     pathfile = await download(url, playlist, current_download_dir)
     filename = pathfile.replace(current_download_dir, "")
 
     # getting the path only
     subdir_path = pathfile.replace(filename, "")
     clean_filename = filename.replace("NA -", "")
+    MEDIA_TYPE = setAudioFormat(media_type)
```
Comment (sylvain-reynaud) [inline app/main.py:33]: why in capitals ?

File: app/main.py [hunk 2]
```diff
@@ -20,21 +22,22 @@ async def read_root():
 
 
 @app.get("/download")
-async def download_audio(url: Optional[str] = None, playlist: Optional[bool] = False):
+async def download_audio(url: Optional[str] = None, playlist: Optional[bool] = False, media_type: Optional[str] = 'webm'):
```
Comment (sylvain-reynaud) [inline app/main.py:25]: In fact, the user can't choose the format, youtube gives a random one. In order the choose the export format, we need to convert it

Comment (LeoDevSecOps) [reply app/main.py:25]: Okay, removed this little feature, will be done in an other PR

File: app/main.py (hunk 1 above)
Comment (LeoDevSecOps) [reply app/main.py]: Dunno

**Pull request merged**